packages/web-app-files/src/helpers/resources.js

```diff
@@ -102,6 +102,80 @@ export function buildResource(resource) {
   }
 }
 
+export function buildSpace(space) {
+  let spaceImageData, spaceReadmeData, spacePermissions
+  let disabled = false
+
+  if (space.special) {
+    spaceImageData = space.special.find((el) => el.specialFolder.name === 'image')
+    spaceReadmeData = space.special.find((el) => el.specialFolder.name === 'readme')
+  }
+
+  if (space.root) {
+    spacePermissions = space.root.permissions
+
+    if (space.root.deleted) {
+      disabled = space.root?.deleted?.state === 'trashed'
+    }
+  }
+
+  return {
+    id: space.id,
+    fileId: '',
+    mimeType: '',
```

Comment:
Do we need those fields, afaik they will never be set

Reply:
Depends on how the respective fields are used throughout the code. Sometimes there are string modifications applied - which don't work on `undefined`. IMO ok to keep them defined and empty, like it's done here. Will be cleaned up anyway once we refactor this to TypeScript.

Reply:
The way I understood @kulmann , these fields should be available to keep they resource object "aligned" with the other resources. Maybe I got it wrong though, @kulmann ? Jan is right, some of these fields will never be set.